celo-org/celo-monorepo
# Phone Number Privacy in Wallet (Part 1)

## Description

- Implement wallet-side changes for phone number privacy
- Add react-native-bls-threshold lib to wallet
- Update ContractKit Attestation wrapper to work with salted hashes
- Add partial support for phone numbers with multiple addresses 
- Remove contact sync code from dappkit as its no longer viable
- Cleanup some other identity related code in the wallet
- De-dupe mocks for payment requests
- Add Redux store migration for the schema changes

This adds the new RN-bls-threshold lib to the wallet, for that see here:
https://github.com/celo-org/react-native-blind-threshold-bls

And that lib is itself based on this rust project:
https://github.com/celo-org/celo-threshold-bls-rs

## Tested

Tested full onboarding in app using the new PGPNP service to generate a salt.

## Related issues

- Fixes https://github.com/celo-org/celo-labs/issues/329
- Facilitates #1777

## Backwards compatibility

No. The phone hashing is different than before, verification now fetches salts before starting, and contact mapping in the app is removed
